pkg/hive/hive.go

```diff
@@ -62,9 +66,24 @@ type Service struct {
 	wg              sync.WaitGroup
 	peersChan       chan pb.Peers
 	sem             *semaphore.Weighted
+	lru             *lru.Cache // cache for unreachable peers
+	bootnode        bool
 }
 
-func New(streamer p2p.StreamerPinger, addressbook addressbook.GetPutter, networkID uint64, logger logging.Logger) *Service {
+func New(streamer p2p.StreamerPinger, addressbook addressbook.GetPutter, networkID uint64, bootnode bool, logger logging.Logger) (*Service, error) {
+
+	var (
+		lruCache *lru.Cache
+		err      error
+	)
+
+	if !bootnode {
```

Comment:
i think it is fine to initialize this even in bootnode mode. the cache size is negligible when empty, and we don't risk any panics on bootnodes